libs/api/repository/src/lib/gn4/gn4-repository.ts

```diff
@@ -165,4 +177,131 @@ export class Gn4Repository implements RecordsRepositoryInterface {
         )
       )
   }
+
+  /**
+   * Returns null if the record is not found
+   */
+  private loadRecordAsXml(uniqueIdentifier: string): Observable<string | null> {
+    return this.gn4RecordsApi
+      .getRecordAs(
+        uniqueIdentifier,
+        undefined,
+        false,
+        undefined,
+        undefined,
+        undefined,
+        'application/xml',
+        'response',
+        undefined,
+        { httpHeaderAccept: 'text/xml,application/xml' as 'application/xml' } // this is to make sure that the response is parsed as text
+      )
+      .pipe(
+        map((response) => response.body),
+        catchError((error: HttpErrorResponse) =>
+          error.status === 404 ? of(null) : throwError(() => error)
+        )
+      )
+  }
+
+  private getLocalStorageKeyForRecord(uniqueIdentifier: string) {
+    return `geonetwork-ui-draft-${uniqueIdentifier}`
+  }
+
+  openRecordForEdition(
+    uniqueIdentifier: string
+  ): Observable<[CatalogRecord, string, boolean] | null> {
+    const draft$ = of(
+      window.localStorage.getItem(
+        this.getLocalStorageKeyForRecord(uniqueIdentifier)
+      )
+    )
+    const recordAsXml$ = this.loadRecordAsXml(uniqueIdentifier)
+    return combineLatest([draft$, recordAsXml$]).pipe(
```

Comment:
good point, originally I made it return an observable because the draft was saved in the backend; I left it like that because I thought it would make sense to allow another implementation of the draft system which would be asynchronous. But of course YAGNI would dictate to get rid of it and make it synchronous... I'll rework it